docs/source/pull_requests.rst

```diff
@@ -0,0 +1,45 @@
+.. _pull_requests:
+
+Pull requests
+*************
+
+Pull requests are one of our most important and powerful development tools.
+They help to obtain good quality, homogenous, and maintainable code.
+
+Both reviewing pull requests and addressing reviewers’ comments is hard.
+The following thoughts might help to obtain an efficient and pleasant reviewing process.
+
+Before issuing a pull request
+-----------------------------
+
```

Comment:
I would add also here something about opening an issue beforehand and communication (see my previous comment about this) in coding_guidelines.rst